### ISSUE TITLE
Impossible remettre un adhérent "auto-désinscrit" en "Inscrit". Problèmatique lorsqu'il y a un paiement en ligne

### ISSUE THREAD

**Issue by fberu** [Club-Alpin-Annecy/collectives#480]
J'ai eu le cas d'un adhérent qui a fait une fausse manip et s'est auto désinscrit.
Actuellement, il ne met pas possible de le remettre facilement en inscrit. Je suis obligé de passer par un "Refusé" puis un "Supprimer l'inscription"

Je me pourrais m'en satisfaire, or cela pose un gros problème lorsqu'il y a un paiement en ligne:

* l'auto désinscription ne supprime pas la paiement: normal
* du coup pour le réinscrire je suis obligé de "bidouiller" une inscription hors ligne à 0€

Objectif du ticket:

* ajouter une transition auto désinscrit -> inscrit
* (si possible) interdire une auto désinscription sur une collective ayant un paiement en ligne

**Comment by gdaviet** [comment]
Ce n’est pas un bug, c’est le comportement voulu.
Premièrement, le coup de la fausse manip semble un peu gros. Il y a une fenêtre de confirmation à la desinscription, cela ressemble plutôt à un changement d’avis. Ce genre de cas ne devrait pas correspondre à une utilisation normale.

Pour rentrer dans le cœur du sujet, il me semble problématique de laisser la possibilité de remettre un desinscrit en inscrit dans vérification manuelle du paiement existant. Les conditions tarifaires peuvent avoir changé entre temps par exemple. ou le paiement peut ne pas avoir été complet. Du coup je tiens à ce que la validation se fasse manuellement par l’encadrant en entrant un paiement hors ligne.

On pourrait envisager de donner plus facilement accès au paiement précédent pour validation. Mais ajouter directement la transition desinscrit -> inscrit, pour moi c’est non.

A noter que tu ne devrais pas être obligé de passer par refuser pour le desinscrite si il s’est auto-desinscrit, on peut supprimer l’inscription directement. Ou passer tout de suite à “en attente de paiement”

**Comment by fberu** [comment]
Je suis d'accord, le coup de la fausse manip est un peu gros. Mais la personne n'a pas su m'expliquer comment elle a pu se retrouver dans ce cas de figure :/

En fait je n'avais pas fait gaf qu'il y a un workflow pour les collectives sans paiement (là on peut faire une transition auto inscrit -> inscrit) et pour les collectives avec paiement.

Ta dernière proposition d'interdire une auto désincription sur les collectives avec paiement me semble parfait :)

**Comment by gdaviet** [comment]
> 
> Ta dernière proposition d'interdire une auto désincription sur les collectives avec paiement me semble parfait :)

Ok, je vais voir pour ajouter ca alors  !

**Comment by jnguiot** [comment]
Donc ca veut dire qu'un participant d'une collective payante qui ne viendrait pas ne pourrait être que refusé par l'encadrant?

**Comment by jnguiot** [comment]
A noter qu'on pourrait embarquer ca dans #690
